libraries/AP_RangeFinder/AP_RangeFinder_Lua.cpp

```diff
@@ -22,20 +22,49 @@
 
 // constructor
 AP_RangeFinder_Lua::AP_RangeFinder_Lua(RangeFinder::RangeFinder_State &_state, AP_RangeFinder_Params &_params) :
-    AP_RangeFinder_Backend(_state, _params)
+    AP_RangeFinder_Backend(_state, _params), _distance_m(0.0f), _signal_quality_pct(-1)
 {
     set_status(RangeFinder::Status::NoData);
 }
 
 
 // Set the distance based on a Lua Script
 bool AP_RangeFinder_Lua::handle_script_msg(float dist_m)
+{
+    return handle_script_msg_ex(dist_m, -1.0f);
+}
+
+
+// Set the distance and signal_quality based on a Lua Script.
+// If the signal_quality_pct argument is an int8_t from lua, then
+// things don't work. Not sure why but using a float works better.
```

Comment:
Hmmm... that's weird. Worth discussing w/ Ryan (I think he was involved in generating the binding code.)